### PR TITLE
Support Lz4 compression

### PR DESCRIPTION
**Pull request by shezadkhan137**

### CODE REVIEW

File: decorators/lz4/receiver.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"github.com/pierrec/lz4/v4"
+	"github.com/zerofox-oss/go-msg"
+)
+
+// Decoder wraps a msg.Receiver with lz4 decoding functionality.
+// It only attempts to decode the Message.Body if Content-Compression
+// is set to lz4.
+func Decoder(next msg.Receiver) msg.Receiver {
+	return msg.ReceiverFunc(func(ctx context.Context, m *msg.Message) error {
+		select {
```
Comment (Xopherus) [inline decorators/lz4/receiver.go:14]: I know this is copied from other parts of this repo, but I think we probably should get in a habit of omitting this select block moving forward. It is almost always going to proceed with the default case. If we want to check for context cancellation first, then I would suggest we have a simple if statement at the top of the function. I think we originally used this thinking there's a race between 2 blocking calls but it's not.

**Review by Xopherus — APPROVED** [review]
LPGTM. Is there anything we need to do to merge or reconcile this against #19?

File: decorators/lz4/receiver.go (hunk 1 above)
Comment (shezadkhan137) [reply decorators/lz4/receiver.go:14]: K, sounds good. Will remove.

**Comment by shezadkhan137** [comment]
@Xopherus I've taken the header name from Kavs PR, and closed it. This one is good to go I think.

**Review by Xopherus — APPROVED** [review]

**Comment by Xopherus** [comment]
Pretty sure the travis integration is bad. We'll have to move to GH actions. In any case I'm going to merge as is

**Pull request merged**